### PR TITLE
Update ruff rules

### PR DESCRIPTION
**Pull request by namurphy**
This PR enables some more ruff rules, and applies some changes and `# noqa` comments to get the rules to pass.

### CODE REVIEW

**Comment by namurphy** [comment]
This is ready for review. The drop in code coverage is because of `ruff` adding explicit `return None` statements that had previously been implicit.  This was only done for functions which had possible return values other than `None`.

I'm planning to create an issue about modifying these functions so that the `return None` isn't a possibility.  In the meantime, it is useful information to have these lines marked as not covered by tests.

File: plasmapy/particles/tests/test_decorators.py [hunk 1]
```diff
@@ -344,10 +343,6 @@ def test_preserving_signature_with_stacked_decorators(decorator1, decorator2):
     assert undecorated_signature == decorated_signature_1_2 == decorated_signature_2_1
 
 
-@pytest.mark.xfail(
-    condition=sys.version_info < (3, 9),
```
Comment (pheuer) [inline plasmapy/particles/tests/test_decorators.py:348]: I assume this passes now?

**Review by pheuer — APPROVED** [review]
LGTM

File: plasmapy/particles/tests/test_decorators.py (hunk 1 above)
Comment (namurphy) [reply plasmapy/particles/tests/test_decorators.py:348]: This was a test failure specific to Python 3.8, and we now require Python ≥3.9, so we're all set with this.

**Comment by namurphy** [comment]
Thank you for the review!

**Pull request merged**